homeassistant/components/bang_olufsen/media_player.py

```diff
@@ -358,6 +462,135 @@ def _async_update_volume(self, data: VolumeState) -> None:
 
         self.async_write_ha_state()
 
+    @callback
+    async def _async_update_name_and_beolink(self) -> None:
+        """Update the device friendly name."""
+        beolink_self = await self._client.get_beolink_self()
+
+        # Update device name
+        device_registry = dr.async_get(self.hass)
+        device_registry.async_update_device(
+            device_id=cast(DeviceEntry, self.device_entry).id,
+            name=beolink_self.friendly_name,
+        )
+
+        await self._async_update_beolink()
+
+    @callback
+    async def _async_update_beolink(self) -> None:
+        """Update the current Beolink leader, listeners, peers and self."""
+
+        self._beolink_attribute = {}
+
+        # Add Beolink self
+        assert self.device_entry
+
+        self._beolink_attribute = {
+            "beolink": {"self": {self.device_entry.name: self._beolink_jid}}
+        }
+
+        # Add Beolink peers
+        peers = await self._client.get_beolink_peers()
+
+        if len(peers) > 0:
+            self._beolink_attribute["beolink"]["peers"] = {}
+            for peer in peers:
+                self._beolink_attribute["beolink"]["peers"][peer.friendly_name] = (
+                    peer.jid
+                )
+
+        self._remote_leader = self._playback_metadata.remote_leader
+
+        # Temp fix for mismatch in WebSocket metadata and "real" REST endpoint where the remote leader is not deleted.
```

Comment:
If this is a temp fix, what is the real fix?

Reply:
I just checked and this seems to have been fixed, so I'll just remove this.